app/models/candidate.rb

```diff
@@ -53,7 +53,12 @@ def touch_application_choices_and_forms
   end
 
   def self.for_email(email)
-    find_or_initialize_by(email_address: email.downcase) if email
+    if (email_address = email&.downcase&.strip).present?
+      joins(:one_login_auth).find_by(one_login_auth: { email_address: email_address }) ||
+        find_or_initialize_by(email_address: email_address)
```

Comment:
```suggestion
      joins(:one_login_auth).find_by(one_login_auth: { email_address: }) ||
        find_or_initialize_by(email_address:)
```

Just a little suggestion from your favourite pedant.